embedchain/embedchain.py

```diff
@@ -129,13 +130,56 @@ def online(self, value):
             raise ValueError(f"Boolean value expected but got {type(value)}.")
         self.llm.online = value
 
+    def check_input(
+        self,
+        source: Any,
+        data_type: DataType,
+        kwargs: Dict[str, Any],
+    ):
+        if data_type == DataType.POSTGRES:
```

Comment:
Is it possible to move this check inside the loader class? If we keep adding these if conditions here, then this file will blow up quickly.

Reply:
Sure